packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts

```diff
@@ -450,23 +468,22 @@ describeNoCompat("GC data store tombstone tests", (getTestObjectProvider) => {
             const container = await loadContainer(summaryVersion);
             await sendOpToUpdateSummaryTimestampToNow(container);
 
-            // Requesting a tombstoned datastore should fail!
-            await assert.rejects(async () => requestFluidObject<ITestDataObject>(container, unreferencedId),
-                (error) => {
-                    const correctErrorType = error.code === 404;
-                    const correctErrorMessage = error.message === `Datastore removed by gc: ${unreferencedId}`;
-                    return correctErrorType && correctErrorMessage;
-                },
-                `Should not be able to retrieve a tombstoned datastore in non-summarizer clients`,
-            );
-            await assert.doesNotReject(
-                async () => requestFluidObject<ITestDataObject>(summarizingContainer, unreferencedId),
-                `Should be able to retrieve a tombstoned datastore in summarizer clients`,
-            );
+            const tombstoneErrorResponse = await containerRuntime_resolveHandle(container, { url: unreferencedId });
```

Comment:
Can you add comments here explaining why this fails and the next request passes.